src/Haskell/Ide/Engine/Plugin/Ormolu.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Haskell.Ide.Engine.Plugin.Ormolu ( ormoluDescriptor ) where
+
+import Control.Monad.IO.Class ( liftIO, MonadIO(..) )
+import Control.Exception
+import Data.Aeson ( Value ( Null ) )
+import qualified Data.Text as T
+import Ormolu
+import Ormolu.Config (defaultConfig)
+import Ormolu.Exception (OrmoluException)
+import Haskell.Ide.Engine.MonadTypes
+import Haskell.Ide.Engine.PluginUtils
+
+ormoluDescriptor :: PluginId -> PluginDescriptor
+ormoluDescriptor plId = PluginDescriptor
+  { pluginId                 = plId
+  , pluginName               = "Ormolu"
+  , pluginDesc               = "A formatter for Haskell source code."
+  , pluginCommands           = []
+  , pluginCodeActionProvider = Nothing
+  , pluginDiagnosticProvider = Nothing
+  , pluginHoverProvider      = Nothing
+  , pluginSymbolProvider     = Nothing
+  , pluginFormattingProvider = Just provider
+  }
+
+provider :: FormattingProvider
+provider contents uri typ _opts =
+  case typ of 
+    -- // TODO Adequately throw an error on the following line
+    FormatRange _ -> return $ IdeResultFail (IdeError PluginError (T.pack $  "ormoluCmd: Selection formatting not supported.") Null)
```

Comment:
Nitpick: this is a user facing error, we should probably make it a bit prettier
```suggestion
    FormatRange _ -> return $ IdeResultFail (IdeError PluginError (T.pack $  "Selection formatting for Ormolu is not currently supported.") Null)
```

Reply:
Ah yes, it's a temporal way of handling it, more as a reminder for me while I work on it. The plan is throw a visible alert to the user.